Adapt unloading for backup integrations

zweckj opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

With #138522 we no longer need to defer the notification of the backup listeners. This cleans up the existing backup integrations.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

zweckj committed:
Adapt unloading for backup integrations

Xitee1 committed:
Fix small typo in qbittorrent strings.json (#138734)

NoRi2909 committed:
Fix user-facing strings of the NWS integration (#138727)

- fix sentence-casing of "API key" to match common string
- remove excessive trailing period from action name
- reword action description to match HA style
- make "Forecast type" description UI-friendly (a selector is available)

zweckj committed:
change order

lellky committed:
Mark reauthentication-flow as exempt for flexit_bacnet (#138740)

lellky committed:
Mark action-exceptions as exempt for flexit_bacnet (#138739)

* Mark action-exceptions as exempt for flexit_bacnet

* Update homeassistant/components/flexit_bacnet/quality_scale.yaml

---------

Co-authored-by: Josef Zweck <josef@zweck.dev>

NoRi2909 committed:
Fix wrong description of teslemetry.set_scheduled_charging action (#138723)

The action allows the user to set a time at which to start charging, but the action's description uses the wrong word "completed".

CFenner committed:
Bump PyViCare to 2.43.1 (#138737)

bump PyViCare to 2.43.1

bdraco committed:
Bump aioesphomeapi to 29.1.0 (#138742)

jbouwh committed:
Fix temp files of mqtt CI tests not cleaned up properly (#138741)

* Fix temp files of mqtt CI tests not cleaned up properly

* Do not cleanup tempfiles, patch gettempdir only

Bre77 committed:
Update Diagnostics in Teslemetry (#138759)

* Testing

* Diag

starkillerOG committed:
Allow playback of h265 encoded Reolink video (#138667)

bdraco committed:
Bump zeroconf to 0.145.1 (#138763)

NoRi2909 committed:
Fixing casing mistakes in user-facing strings of renault (#138729)

- use sentence-casing for strings
- use uppercase for "ID"

NoRi2909 committed:
Make spelling of "BSB-Lan" consistent (#138766)

abmantis commented on homeassistant/components/synology_dsm/__init__.py:
```diff
@@ -131,18 +131,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Synology DSM sensors."""
+    _async_notify_backup_listeners(hass)
```
shouldn't the listener be notified after the platforms are unloaded? (same for the other backup integrations)

zweckj replied:
Discussed that with Erik, it shouldn’t really make a difference, but the listeners come first as they can’t fail

abmantis commented on homeassistant/components/google_drive/__init__.py:
```diff
@@ -49,7 +49,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: GoogleDriveConfigEntry)
         raise ConfigEntryNotReady from err
 
-    _async_notify_backup_listeners_soon(hass)
+    _async_notify_backup_listeners(hass)
```
I was now wondering if this may trigger the listener to call `async_get_backup_agents` before the this method finishes. if so, is the entry in `hass.config_entries.async_loaded_entries(DOMAIN)` at that point?

mib1185 replied:
I think the same counts for synology dsm integartion - we had the issue, that the `async_get_backup_agents` gets called to early, the config entry was not unloaded at this moment, so was still considered as a valid backup agent

zweckj replied:
No, the unloading part is fixed. The question is during setup.

mib1185 replied:
ah ok 👍 but (_at least for the synology dsm_) we rely on the fact, that a config entry needs to be in state loaded, so when the `async_get_backup_agents` before the config entry is set to loaded, it will fail to load the backup agent 🤔

zweckj replied:
yes, it does work practically. The question is why 😅

mib1185 replied:
maybe because the `async_get_backup_agents` is async and the last action in the `async_setup_entry`, the `async_setup_entry` might be finished first, before any of the backup agent listeners gets fired 🤔

abmantis replied:
I assume it works because the listener uses `async_create_task` to call `async_get_backup_agents`, which makes the call happen after `async_setup_entry`. Still, we should probably change this here and not depend on the internals of the backup domain.

nivstein committed:
Bump pyrympro from 0.0.8 to 0.0.9 (#138753)

emontnemery committed:
Don't allow setting backup retention to 0 days or copies (#138771)

* Don't allow setting backup retention to 0 days or copies

* Add tests

MindFreeze committed:
Fix Z-WaveJS inclusion in the background (#138717)

* Fix Z-WaveJS inclusion in the background

* improve async handling

* just return the `requested_grant` to the driver

* handle controller busy state

PeteRager committed:
Fix TV input source option for Sonos Arc Ultra (#138778)

initial commit

NoRi2909 committed:
Fix grammar in evohome.reset_system action, consistently add "mode" (#138777)

* Fix grammar in evohome.reset_system action, consistently add "mode"

- fix the grammar with "Sets … and resets …"
- add "mode" to all mode names for consistency

* Revert, removing one excessive "mode"

peteS-UK committed:
Add support for announce to Squeezebox media player (#129460)

* initial

* Add support for announce: true to media player

* Move play_announcement to _player

* update snapshot

* conftest update

* remove conftest update

* Update conftest.py

* Test Updates

* Updates post moving functions to library

* test fixes

* Review updates

* Snapshot update

* rebase updates

* Merge updates

* Review updates

* Review updates

andrewsayre committed:
Add HEOS entity service to set group volume level (#136885)

Sibgatulin committed:
airq: add more verbose debug logging (#138192)

parkerbxyz committed:
Add threshold sensor to Aranet (#137291)

* Add threshold level sensor description to Aranet component

* Use Color enum for status options

* Add threshold level sensor tests for Aranet components

* Rename `threshold_level` key to `status`

* Update test to expect 7 sensors instead of 6

* Map sensor status to more human-friendly strings

* Rename `threshold_level` key to `concentration_status`

* Update docstring for  function

* Simplify `get_friendly_status()`

* Rename `concentration_status` to `concentration_level`

* Rename `concentration_status` to `concentration_level` in sensor tests

* Refactor concentration level handling and tests

* Normalize concentration level status values to lowercase

* Add error to translations

* Don't scale status string

* Apply suggestions from code review

Co-authored-by: Shay Levy <levyshay1@gmail.com>

* Rename `concentration_level` to `threshold_indication`

* Update threshold indication translations

* `threshold_indication` → `threshold`

* Capitalize sensor name

Co-Authored-By: Shay Levy <levyshay1@gmail.com>

---------

Co-authored-by: Shay Levy <levyshay1@gmail.com>

matrixd2 committed:
Add switch flex button support. (#137524)

bieniu committed:
Add `_shelly._tcp` to Shelly zeroconf configuration (#138782)

Add _shelly._tcp to zeroconf

SLaks committed:
Ecobee: Report Humidifier Action (#138756)

Co-authored-by: Josef Zweck <josef@zweck.dev>

edenhaus committed:
Bump uv to 0.6.1 (#138790)

edenhaus committed:
Reorder Dockerfile to improve caching (#138789)

bdraco committed:
Fix shelly not being able to be setup from user flow when already discovered (#138807)

raise_on_progress=False was missing in the user flow which
made it impossible to configure a shelly by IP when there
was an active discovery because the flow would abort

Diegorro98 committed:
Add Home Connect to .strict-typing (#138799)

* Add Home Connect to .strict-typing

* Fix mypy errors

Diegorro98 committed:
Add Home Connect entities that weren't added before (#138796)

Added entities that weren't added before

Diegorro98 committed:
Set Home Connect beverages counters as diagnostics (#138798)

Set beverages counters as diagnostics

Diegorro98 committed:
Add connectivity binary sensor to Home Connect (#138795)

Add connectivity binary sensor

frenck committed:
Update Home Assistant base image to 2025.02.1 (#138746)

* Update Home Assistant base image to 2025.02.1

* Require Python 3.13.2 now

CFenner committed:
Fix typos in strings.json files (#138601)

* fix codespell issues

* update nextcloud snapshots

* update weheat snapshots

* update waqi snapshots

andrewsayre committed:
Add HEOS group volume down/up actions (#138801)

Add group volume down/up actions

skobow committed:
Set  clean_start=True on connect to MQTT broker (#136026)

* Addresses #135443: Set  on connect.

* Make clean start implementation compatible with v2 API

* Add tests

* Do not pass default value for `clean_start` on_connect

* Revert "Do not pass default value for `clean_start` on_connect"

This reverts commit 75806736cf511a6d6b6496454843de34f05f7758.

* Use partial top pass kwargs to mqtt client connect

---------

Co-authored-by: Jan Bouwhuis <jbouwh@users.noreply.github.com>
Co-authored-by: jbouwh <jan@jbsoft.nl>

synesthesiam committed:
Add assistant filter to expose entities list command (#138817)

bdraco committed:
Bump aioesphomeapi to 29.1.1 (#138827)

jbouwh committed:
Move blocking code to executor job in MQTT CI test helper (#138815)

bdraco committed:
Bump habluetooth to 3.22.0 (#138812)

HA-Roberto committed:
Add button for bond light temp toggle feature (#135379)

Co-authored-by: J. Nick Koston <nick@koston.org>

mikey0000 committed:
Electric Kiwi: Add quality scale (#138680)

* add quality scale file

* Apply suggestions from code review

Co-authored-by: Josef Zweck <josef@zweck.dev>

* add suggestions and add extra missing icon

* update a few based on documentation

* exempt installation parameters

* set a few more documentation items to done

* Update homeassistant/components/electric_kiwi/quality_scale.yaml

Co-authored-by: Josef Zweck <josef@zweck.dev>

* update reason for no installation parameters

* set docs installation parameters to done

* revert back to exempt

* add bronze scale

---------

Co-authored-by: Josef Zweck <josef@zweck.dev>

mikey0000 committed:
add icon to select (#138834)

lellky committed:
Add test for flexit_bacnet hvac mode (#138748)

Add test for hvac mode

bdraco committed:
Improve performance of calculating state (#138832)

```
print(timeit.timeit("x.update(y)", setup=x={a:b}

CFenner committed:
Add sensors for pellets boiler in ViCare integration (#138563)

* add buffer sensors

* remove duplicate sensor

* add labels

* Bump PyViCare to 2.43.0

* add fuel need sensor

mikey0000 committed:
Electric Kiwi: Parallel updates (#138839)

* parallel updates

* Update homeassistant/components/electric_kiwi/select.py

lellky committed:
Remove unused code in the climate entity of the flexit_bacnet integration (#138840)

Removes unused code in the climate entity

This was unintentionally left in the code when adding a coordinator

farmio committed:
Update xknx to 3.6.0 (#138838)

LG-ThinQ-Integration committed:
Modify string water_heater's off state (#137627)

* Modify string water_heater's off state

* Modify washer's delay name

---------

Co-authored-by: yunseon.park <yunseon.park@lge.com>

ludeeus committed:
Add check_connection parameter to cloud login methods and handle AlreadyConnectedError (#138699)

proohit committed:
Adds Tado Child Lock support (#135837)

edenhaus committed:
Add WIND_DIRECTION to SensorDeviceClass and NumberDeviceClass (#138714)

* Add WIND_DIRECTION to SensorDeviceClass

* Add WIND_DIRECTION to NumberDeviceClass

* Fix tests

starkillerOG committed:
Fix playback for encrypted Reolink files (#138852)

emontnemery committed:
Clean up translations for mocked integrations inbetween tests (#138732)

* Clean up translations for mocked integrations inbetween tests

* Adjust code, add test

* Fix docstring

* Improve cleanup, add test

* Fix test

zweckj commented:
Replaced by #138858
